examples/solvers/chambolle_pock_denoising_tgv.py

```diff
@@ -0,0 +1,151 @@
+"""Total generalized variation denoising using the Chambolle-Pock solver.
+
+Solves the optimization problem
+
+    min_x ||x - d||_2^2 + TGV_2(x)
+
+Where ``d`` is given noisy data TGV_2 is the second order total generalized
+variation of ``x``, defined as
+
+    TGV_2(x) = min_y lam_1 ||Gx - y||_1 + lam_2 ||Ey||_1
+
+where ``G`` is the spatial gradient operating on the scalar-field ``x`` and
+``E`` is the symmetrized gradient operating on the vector-field ``y``.
+Both one-norms ||.||_1 take the one-norm globally and the two-norm locally,
+i.e. ||y||_1 := sum_i sqrt(sum_j y_i(j)^2) where y_i(j) is the j-th value
+of the vector y_i at location i.
+
+The problem is rewritten as
+
+    min_{x, y} ||x - d||_2^2 +  lam_1 ||Gx - y||_1 + lam_2 ||Ey||_1
+
+which can then be solved with the Chambolle-Pock method.
+
+References
+----------
+[1] K. Bredies and M. Holler. A TGV-based framework for variational image
+decompression, zooming and reconstruction. Part II: Numerics. SIAM Journal
+on Imaging Sciences, 8(4):2851-2886, 2015.
+"""
+
+import numpy as np
+import odl
+
+
+# %%
```

Comment:
done. Actually, those are no left-overs. These create boxes in spyder which can be executed one by one for an interactive example.